File: SocialPulse/SocialPulse.Core/Entities/Image.cs
namespace SocialPulse.Core
{
    public class Image : BaseEntity
    {
        public byte[] Data { get; set; } = null!;
        public string ContentType { get; set; } = null!;

        public int? PostId { get; set; }
        public Post? Post { get; set; }
        public int? MessageId { get; set; }
        public Message? Message { get; set; }
    }
}

File: SocialPulse/UI/admin_socialpulse/lib/models/image.dart
import 'package:json_annotation/json_annotation.dart';

part 'image.g.dart';

@JsonSerializable()
class Image{
  int? id;
  String? data;
  String? contentType;
  int? postId;
  int? messageId;

  Image (this.id,this.data,this.contentType,this.postId,this.messageId);

  factory Image.fromJson(Map<String, dynamic> json) =>
      _$ImageFromJson(json);

  Map<String, dynamic> toJson() => _$ImageToJson(this);
}